### PR TITLE
PrivateContract: enable constructor call

### PR DESCRIPTION
**Pull request by pcppcp**
required for the deployment script

### CODE REVIEW

**Comment by rakanalh** [comment]
Reviewing this

File: raiden_libs/private_contract.py [hunk 1]
```diff
@@ -59,8 +67,12 @@ def __init__(self, contract):
         self.contract = contract
         self.functions = FunctionsMap(self.contract.abi, contract)
 
+    def constructor(self, *args, **kwargs):
```
Comment (rakanalh) [inline raiden_libs/private_contract.py]: Can you add return type here?

File: raiden_libs/private_contract.py [hunk 2]
```diff
@@ -11,13 +15,17 @@ def __init__(self, call):
...
         """Creates a transaction, signs it with a provided private key and sends it. """
+        args = deepcopy(args)
         private_key = kwargs.pop('private_key', None)
         if len(args) == 0:
             args = ({},)
         # buildTransaction requires 'from' field to be set properly
         args[0]['from'] = private_key_to_address(private_key)
         gas_limit = args[0].pop('gas_limit', GAS_LIMIT_CONTRACT)
-        tx_data = self._call.buildTransaction(*args, **kwargs)
+        if isinstance(self._call, ContractConstructor):
```
Comment (rakanalh) [inline raiden_libs/private_contract.py]: Please add some docs as to why we're distinguishing between how we call constructor from other methods?

Comment (pcppcp) [reply raiden_libs/private_contract.py]: actually it works fine without the switch... :)

**Pull request merged**